### PR TITLE
CDAP-2447 adapter config can have no transforms property

### PR DESCRIPTION
**Pull request by gokulavasan**
Can contain only source and sink if no transforms are required.

Build : http://builds.cask.co/browse/CDAP-DT97

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/ETLConfig.java [hunk 1]
```diff
@@ -44,8 +45,8 @@ public ETLStage getSink() {
     return sink;
   }
 
-  public List<ETLStage> getTransforms() {
-    return transforms;
+  public Optional<List<ETLStage>> getTransforms() {
+    return Optional.fromNullable(transforms);
```
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/ETLConfig.java]: Do we need to return an Optional? Can it be just `return transforms == null ? new ArrayList<>() : transforms;`?

Comment (gokulavasan) [inline cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/ETLConfig.java]: Yes, that will work too. Will update the PR.

**Comment by gokulavasan** [comment]
@chtyim Build passed. Please take a look. Thanks!

**Comment by chtyim** [comment]
LGTM

**Pull request merged**